pkg/cmd/issue/comment/comment.go

```diff
@@ -0,0 +1,191 @@
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Config:     f.Config,
+		Edit: func(editorCommand string) (string, error) {
+			return surveyext.Edit(editorCommand, "*.md", "", f.IOStreams.In, f.IOStreams.Out, f.IOStreams.ErrOut, nil)
+		},
+	}
+
+	var webMode bool
+	var editorMode bool
+
+	cmd := &cobra.Command{
+		Use:   "comment {<number> | <url>}",
+		Short: "Create a new issue comment",
+		Example: heredoc.Doc(`
+			$ gh issue comment 22 --body "I was able to reproduce this issue, lets fix it."
+		`),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+			opts.SelectorArg = args[0]
+
+			inputFlags := 0
+			if cmd.Flags().Changed("body") {
+				opts.InputType = inline
+				inputFlags++
+			}
+			if webMode {
+				opts.InputType = web
+				inputFlags++
+			}
+			if editorMode {
+				opts.InputType = editor
+				inputFlags++
+			}
+
+			if inputFlags == 0 {
+				if !opts.IO.CanPrompt() {
+					return &cmdutil.FlagError{Err: errors.New("--body or --web required when not running interactively")}
+				}
+				opts.Interactive = true
+			} else if inputFlags == 1 {
+				if !opts.IO.CanPrompt() && opts.InputType == editor {
+					return &cmdutil.FlagError{Err: errors.New("--body or --web required when not running interactively")}
+				}
+			} else if inputFlags > 1 {
+				return &cmdutil.FlagError{Err: fmt.Errorf("specify only one of --body, --editor, or --web")}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return commentRun(opts)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.Body, "body", "b", "", "Supply a body. Will prompt for one otherwise.")
+	cmd.Flags().BoolVarP(&editorMode, "editor", "e", false, "Add body using editor")
+	cmd.Flags().BoolVarP(&webMode, "web", "w", false, "Add body in browser")
+
+	return cmd
+}
+
+func commentRun(opts *CommentOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+	apiClient := api.NewClientFromHTTP(httpClient)
+
+	issue, baseRepo, err := shared.IssueFromArg(apiClient, opts.BaseRepo, opts.SelectorArg)
+	if err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		inputType, err := inputTypeSurvey()
+		if err != nil {
+			return err
+		}
+		opts.InputType = inputType
+	}
+
+	switch opts.InputType {
+	case web:
+		openURL := issue.URL + "#issuecomment-new"
+		if opts.IO.IsStdoutTTY() {
+			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(openURL))
+		}
+		return utils.OpenInBrowser(openURL)
+	case editor:
+		editorCommand, err := cmdutil.DetermineEditor(opts.Config)
+		if err != nil {
+			return err
+		}
+		body, err := opts.Edit(editorCommand)
+		if err != nil {
+			return err
+		}
+		opts.Body = body
+		if opts.Interactive {
+			cs := opts.IO.ColorScheme()
+			fmt.Fprint(opts.IO.Out, cs.GreenBold("?"))
+			fmt.Fprint(opts.IO.Out, cs.Bold(" Body "))
+			fmt.Fprint(opts.IO.Out, cs.Cyan("<Received>\n"))
+		}
```

Comment:
I ended up removing the extra survey that prompts the user to enter "e" to open their editor but that also removed the confirmation output. I liked the little confirmation that `gh` had received the body from the editor so I recreated it here, keeping the same survey style.

Reply:
Props for the extra effort, but I don't think that `? Body <Received>` communicates anything useful to the user. I don't think that we need to maintain this logic.